framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -2040,9 +2035,44 @@ public Void visitNewClass(NewClassTree node, Void p) {
         AnnotatedExecutableType constructorType = fromUse.executableType;
         List<AnnotatedTypeMirror> typeargs = fromUse.typeArgs;
 
+        // Type check inner class receiver type
+        AnnotatedTypeMirror parameterReceiverType = constructorType.getReceiverType();
+        AnnotatedTypeMirror argumentReceiverType = null;
```

Comment:
```suggestion
        AnnotatedTypeMirror argumentReceiverType;
```

Don't unnecessarily initialize local variables.